util/git/creds.go

```diff
@@ -33,19 +33,24 @@ func (c NopCreds) Environ() (io.Closer, []string, error) {
 
 // HTTPS creds implementation
 type HTTPSCreds struct {
-	username, password string
+	username, password    string
+	insecureSkipTLSVerify bool
 }
 
-func NewHTTPSCreds(username, password string) HTTPSCreds {
-	return HTTPSCreds{username, password}
+func NewHTTPSCreds(username, password string, insecureSkipTLSVerify bool) HTTPSCreds {
+	return HTTPSCreds{username, password, insecureSkipTLSVerify}
 }
 
 func (c HTTPSCreds) Environ() (io.Closer, []string, error) {
-	return NopCloser{}, []string{
+	env := []string{
 		fmt.Sprintf("GIT_ASKPASS=%s", "git-ask-pass.sh"),
 		fmt.Sprintf("GIT_USERNAME=%s", c.username),
 		fmt.Sprintf("GIT_PASSWORD=%s", c.password),
-	}, nil
+	}
+	if c.insecureSkipTLSVerify {
+		env = append(env, "GIT_SSL_NO_VERIFY=true")
```

Comment:
this is all about credentials, I'm not sure this is the right file for this flag

Reply:
waiting for @jannfis to complete https://github.com/argoproj/argo-cd/pull/1807, since he started working on it first.